Create coverage report in Travis.

mauritsvanrees opened the pull request:
This should solve issue #93, in a slightly different way than my last proposal there.

Generate nosetests script also under the 'test' name.  Then
bin/createcoverage works.

In Travis call the new correct script names:

- The new bin/test is the old bin/nosetests.
- The new bin/test-plone_addon is the old bin/test.

Created bin/alltests script.  This runs bin/test, and the two
test-plone_addon scripts.  We could call that in Travis too.

Added initial simple test to show coverage in hooks.py.

mauritsvanrees commented:
Travis is expected to fail, see a different pull request for the plone5 /plone.app.event stuff.
But after it is done, coveralls should hopefully get some data.

mauritsvanrees commented:
Yes!  The Plone 5 builds work and they pass info to coveralls.
The Plone 4 builds fail, so they do not contact coveralls, but we know that they fail.
Ready for review.

tisto commented on .travis.yml:
```diff
@@ -39,8 +39,9 @@ before_script:
   - bin/code-analysis
-  - bin/nosetests
   - bin/test
+  - bin/createcoverage
```
I think createcoverage should run in the after_success sestion. This is not a test.

mauritsvanrees replied:
Problem is that it then runs after `bin/test-plone_addon`, which leaves behind a `test.plone_addon` directory that the nosetests then also get run on, causing ImportErrors. Ah, but of course this directory only gets left behind when the command fails, so then there is no need to call createcoverage.  I will move it.

In that case, we should move createcoverage in the generated package too. I will move that too.

tisto commented on all_tests.template.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Personally I prefer to keep such small scripts inline in the buildout. Inline templates are much easier to understand. Otherwise people have to look into many different files to understand what goes on.

mauritsvanrees replied:
Ah, true. Will do.

tisto commented on full_test.py.template:
```diff
@@ -0,0 +1,3 @@
+import subprocess
```
Can you elaborate where this file is actually used?

mauritsvanrees replied:
Oops, this was a left-over from an earlier try. Removed now.

tisto commented:
@mauritsvanrees would you prefer to fix the Plone 4 build in this branch or shall we merge first?

mauritsvanrees commented:
Plone 4 fixes have been merged to master.
I have fixed the issues you raised on this pull request, rebased on master, and did a force push, cleaning things up. Let's see what Travis and Coveralls think of it.

mauritsvanrees commented:
Two minor changes added. Let me stop now. :-)

tisto commented:
Thanks so much @mauritsvanrees! Great work! Will do a release soon.
